public/pages/Emails/components/tables/RecipientGroupsTable.tsx

```diff
@@ -134,6 +136,9 @@ export class RecipientGroupsTable extends Component<
     if (!_.isEqual(prevQuery, currQuery)) {
       await this.refresh();
     }
+    if (isDataSourceChanged(this.props, prevProps)) {
```

Comment:
Nitpick:
This check could be worked into line check on line 125 by changing the check to
```
if (!_.isEqual(prevQuery, currQuery) || isDataSourceChanged(this.props, prevProps)) { ...